File: output/ChatGPT/avatar/Java/Python/atcoder_AGC021_B.py
import math

N = int(input())
x = []
y = []

for i in range(N):
    x_i, y_i = map(int, input().split())
    x.append(x_i)
    y.append(y_i)

for i in range(N):
    thetas = []

    for j in range(N):
        if i == j:
            continue
        thetas.append(math.atan2(y[j] - y[i], x[j] - x[i]))

    thetas.sort()
    thetas.append(thetas[0] + 2 * math.pi)
    ans = 0

    for k in range(N - 1):
        ans = max(ans, thetas[k + 1] - thetas[k] - math.pi)

    print(ans / (math.pi * 2))

File: output/ChatGPT/avatar/Java/C++/atcoder_AGC021_B.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

int main() {
    int N;
    std::cin >> N;
    std::vector<int> x(N);
    std::vector<int> y(N);
    
    for (int i = 0; i < N; ++i) {
        std::cin >> x[i] >> y[i];
    }
    
    for (int i = 0; i < N; ++i) {
        std::vector<double> thetas;
        
        for (int j = 0; j < N; ++j) {
            if (i == j) continue;
            thetas.push_back(std::atan2(y[j] - y[i], x[j] - x[i]));
        }
        
        std::sort(thetas.begin(), thetas.end());
        thetas.push_back(thetas[0] + 2 * M_PI);
        double ans = 0;
        
        for (int k = 0; k < N - 1; ++k) {
            ans = std::max(ans, thetas[k + 1] - thetas[k] - M_PI);
        }
        
        std::cout << ans / (M_PI * 2) << std::endl;
    }

    return 0;
}